vault/auth.go

```diff
@@ -185,11 +185,14 @@ func (c *Core) enableCredentialInternal(ctx context.Context, entry *MountEntry,
 		return fmt.Errorf("cannot mount %q of type %q as an auth backend", entry.Type, backendType)
 	}
 	// update the entry running version with the configured version, which was verified during registration.
-	entry.RunningVersion = entry.Version
 	if entry.RunningVersion == "" {
```

Comment:
What if someone upgrades from v1.0.0 to v1.0.1? I don't think we'll successfully update the running version to v1.0.1 in that case.

I would be inclined to prevent anyone from populating `entry.Version` with an invalid value in the first place. i.e. when it's getting set to anything other than `""`, verify that the plugin name is not one of the disallowed plugins. Then the logic here can be pretty simple and just trust that the `entry.Version` was used faithfully when creating the backend. That should also clean up reads on the mount table, e.g. `vault plugin read sys/mounts/secret` (assuming it's a KV engine) should never report a non-empty value for `plugin_version`.

Reply:
Good call. I moved the logic earlier, which allowed me to cleanup the version validation a little even.